### PR TITLE
Refactoring user story 2 - Krzysztof & Meryl code review

### PR DESCRIPTION
**Pull request by RastallM**

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -14,6 +14,8 @@ class BnbManager < Sinatra::Base
 
   get '/spaces' do
     @result = Space.all
+    ## Before, result looked like: ["room in rotherhithe", "room in westiminster"]
+    ## Now, result looks like: [space_object_1, space_object_2]
     p @result
```
Comment (RastallM) [inline app.rb]: Can we delete this now?

File: app.rb [hunk 2]
```diff
@@ -14,6 +14,8 @@ class BnbManager < Sinatra::Base
 
   get '/spaces' do
     @result = Space.all
+    ## Before, result looked like: ["room in rotherhithe", "room in westiminster"]
+    ## Now, result looks like: [space_object_1, space_object_2]
```
Comment (RastallM) [inline app.rb]: K & M understood - we can now remove

File: db/migrations/02_add_desc_to_spaces.sql [hunk 3]
```diff
@@ -0,0 +1 @@
+ALTER TABLE spaces ADD COLUMN description VARCHAR(200);
```
Comment (RastallM) [inline db/migrations/02_add_desc_to_spaces.sql]: We have also added a "02_.." migration file. Lets discuss whether this should just be added to the 01_create_spaces_table.sql file? The reason for this file is for end users to create required tables - its not clear they will need the incremental steps - rather just the end table requirements.

File: spec/features/adding_spaces_spec.rb [hunk 4]
```diff
@@ -5,8 +5,10 @@
     expect(current_path).to eq "/spaces/new"
 
     fill_in :space_name, with: "Cozy Cottage"
+    fill_in :description, with: "Nice walks near by"
     click_button("Submit")
 
     expect(page).to have_content("Cozy Cottage")
+    expect(page).to have_content("Nice walks near by")
   end
```
Comment (RastallM) [inline spec/features/adding_spaces_spec.rb:13]: K&M understood - good to remove comments now - discuss whether comments should be left in for future developers?

File: spec/unit/space_spec.rb [hunk 5]
```diff
@@ -3,22 +3,24 @@
...
+      new_space = Space.create(name: 'Private room rotherhithe', description: 'Close to tube station')
+      Space.create(name:'Entire apartment in westminster', description: 'A view to Big Ben')
+      Space.create(name: 'Private room westminster', description: 'High ceiling')
+      
+      all_spaces = Space.all
+      expect(all_spaces).to be_a Array
+      expect(all_spaces.length).to eq 3
+      expect(all_spaces.first).to be_a Space 
+      expect(all_spaces.first.id).to eq new_space.id
+      expect(all_spaces.first.name).to eq "Private room rotherhithe"
```
Comment (RastallM) [inline spec/unit/space_spec.rb:15]: should line 15 be similar to line 14 in that it should eq new_space.name?

File: views/spaces/new.erb [hunk 6]
```diff
@@ -1,4 +1,5 @@
 <form action='/spaces' method='post'>
   <input type='text' name='space_name' placeholder='Name your space'>
+  <input type='text' name='description' placeholder='Description'>
   <input type='submit' value='Submit'>
```
Comment (RastallM) [inline views/spaces/new.erb]: We have used labels and placeholders for our forms - we should rackup and take a look at each form and decide as a group which one we want to go with and ensure consistency across forms

File: views/spaces/index.erb [hunk 7]
```diff
@@ -1,5 +1,5 @@
-<% @result.each do |name|%>
-  <%=name%>
+<% @result.each do |space_object|%>
+  <%=space_object.name%><%=space_object.description%>
 <%end%>
 
 <form action='/spaces/new' method='get'>
```
Comment (RastallM) [inline views/spaces/index.erb:10]: need to add a section to the form to include the space description - added to trello board

**Review by emilyalice2708 — APPROVED** [review]

**Pull request merged**